### ISSUE TITLE
Rewrite text in metadata display

### ISSUE THREAD

**Issue by bondjimbond** [bondjimbond/islandora_metadata_extras#17]
Some sites want to display custom terms to the users while wanting to retain authoritative vocabularies in their metadata, for use with harvesters and interoperability reasons. It would be useful to have a metadata display text replacement feature...

e.g. A site using the roleTerm "Thesis Advisor" wants to only display the term "Advisor".

Implementation could be a simple text area - original metadata entry on left, replacement text on right, separated by a pipe

e.g.

> Thesis Advisor|Advisor

Should rewrite that string in the output of any Solr field in both search results and metadata display.

**Comment by mjordan** [comment]
Nice feature. Could be accomplished similar to the way that yyyy-mm-dd dates are converted to words.

**Comment by mjordan** [comment]
@bondjimbond I'll point out that one UX complication to rewriting output in this way is that a user might assume (naturally so) that the rewritten value she sees in the metadata display is something she can then search for. Since the rewritten output is not indexed, searches will not return what the user expects. Also, facets will use the indexed version of the value, not the rewritten one.

**Comment by bondjimbond** [comment]
This is true. I imagine it's not a big deal for something like a role term, since they'd more likely search for the person's name. Facets are another beast, though...

**Comment by mjordan** [comment]
I definitely think it's worth implementing, but we should point out the gotchas. In fact, I should do that for dates already.

Do you want me to take a first crack at it?

**Comment by MarcusBarnes** [comment]
It may be possible to overcome possible UX confusion with regards to search for the display value rather than the indexed value using Solr's ability to deal with synonyms.  This would require additional manual Solr configuration, but it would address the [UX complication noted](https://github.com/mjordan/islandora_metadata_extras/issues/17#issuecomment-360600615) that would result from adding this feature.

**Comment by mjordan** [comment]
Good point!

**Comment by bondjimbond** [comment]
Thanks, @mjordan - would be happy to see you take a shot at it. I'll test when you've got something.

**Comment by mjordan** [comment]
OK, I'll do up a proof of concept.
